src/js/admin-settings.ts

```diff
@@ -0,0 +1,25 @@
+document.querySelector( '.media-single-image button.browse' )?.addEventListener( 'click', selectImage );
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+function selectImage( this: any ) {
+	const optionName = this.dataset.option;
```

Comment:
Lots of ways to go about this, but recommend using event param instead of this and injecting the correct DOM type:
```suggestion
function selectImage( evt: Event ) {
	const optionName = ( evt.target as HTMLButtonElement ).dataset.option;
```